test/test-network/systemd-networkd-tests.py

```diff
@@ -382,13 +382,18 @@ def wait_online(self, links_with_operstate, timeout='20s', bool_any=False, setup
                 output = check_output(*networkctl_cmd, 'status', link.split(':')[0], env=env)
                 print(output)
             raise
-        if not bool_any:
-            for link in links_with_operstate:
-                output = check_output(*networkctl_cmd, 'status', link.split(':')[0])
-                print(output)
-                for line in output.splitlines():
-                    if 'State:' in line:
-                        self.assertRegex(line, setup_state)
+        if not bool_any and setup_timeout:
```

Comment:
This does not look right, the condition should still be checked even if setup_timeout is unset.

Reply:
ok i changed it to check `not bool_any and setup_state` (since the check requires setup_state to be set to something) and changed the for loop to use `range(setup_timeout or 1)` so it will check at least once.